### ISSUE TITLE
Segmentation fault when trying to read contents of an empty file using `file.text()`

### ISSUE THREAD

**Issue by Xkonti** [oven-sh/bun#4406]
### What version of Bun is running?

0.8.1+16b4bf341acc0f4804f0b6bdf5298c180cd00366

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

Reading contents of an empty (yet existing) file using the `file.text()` causes the `Segmentation fault`.
Running the following test:

```ts
test('should return a file reference if the file is found', async () => {
  const path = 'testdata/emptiness.txt';
  const file = Bun.file(path); // ✔
  expect(file?.size).toBe(0); // ✔
  console.log('Checking file contents: ', path);
  const contents = await file?.text(); // ❌
  console.log('File contents are: ', contents);
  expect(contents).toBe('');
});
```

Causes the test not to finish with the following output:
```
src/utils/fileUtils.test.ts:
Checking file contents:  testdata/emptiness.txt
Segmentation fault
```

The file was created programmatically using:
```ts
const path = 'testdata/emptiness.txt';
const contents = '';
await Bun.write(path, contents);
```

The test is successfully passing files that have contents set to some non-empty string. It only fails when it's empty.

On version 0.7.3 of Bun the test was passing correctly.

**Comment by Jarred-Sumner** [comment]
I suspect this is a bug impacting linux kernel versions < 5.16

Can you try upgrading to the latest WSL2 kernel in Windows update?

**Comment by Xkonti** [comment]
@Jarred-Sumner Hmm... I just updated Bun from 0.7.3 to 0.8.1 on a different computer that is also running on Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64. And on that computer all tests work fine.

**Comment by Jarred-Sumner** [comment]
This was fixed in v1.0.1

Duplicate of https://github.com/oven-sh/bun/issues/4272
